pkg/common/kafka/consumer/consumer.go

```diff
@@ -1,66 +1,43 @@
 package consumer
 
 import (
-	"errors"
-	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/Shopify/sarama"
+	"github.com/rcrowley/go-metrics"
 	"knative.dev/eventing-kafka/pkg/common/kafka/constants"
 	"knative.dev/eventing-kafka/pkg/common/kafka/util"
 )
 
-// Confluent Client Doesn't Code To Interfaces Or Provide Mocks So We're Wrapping Our Usage Of The Consumer For Testing
-type ConsumerInterface interface {
-	Subscribe(topic string, rebalanceCb kafka.RebalanceCb) error
-	Poll(int) kafka.Event
-	CommitMessage(*kafka.Message) ([]kafka.TopicPartition, error)
-	Close() error
-	OffsetsForTimes(times []kafka.TopicPartition, timeoutMs int) (offsets []kafka.TopicPartition, err error)
-	GetMetadata(topic *string, allTopics bool, timeoutMs int) (*kafka.Metadata, error)
-	CommitOffsets(offsets []kafka.TopicPartition) ([]kafka.TopicPartition, error)
-	QueryWatermarkOffsets(topic string, partition int32, timeoutMs int) (low, high int64, err error)
-	StoreOffsets(offsets []kafka.TopicPartition) (storedOffsets []kafka.TopicPartition, err error)
-	Commit() ([]kafka.TopicPartition, error)
-}
+// Create A Kafka ConsumerGroup (Optional SASL Authentication)
+func CreateConsumerGroup(clientId string, brokers []string, groupId string, username string, password string) (sarama.ConsumerGroup, metrics.Registry, error) {
```

Comment:
you might find useful the code in `eventing-contrib/kafka/common`

Reply:
Yeah, I definitely looked at some of that code to help build my implementation and in retrospect I wish I had followed the factory pattern a little more closely.  We are returning the metrics Registry from our "factory", and don't pass in the handler, etc - so I can't switch over at this point without a significant rework.  FWIW - I'm also not too happy about our current `/common/kafka` implementation of the kafka resources and am open to a redesign if/when we align/replace the two implementations.  Right now though it's not the top priority.